
import java.awt.*;
import javax.swing.*;
import java.lang.*;
import java.io.*;
import java.util.Random;
import javax.swing.JFileChooser;

import org.jfree.chart.ChartFactory;  
import org.jfree.chart.JFreeChart;  
import org.jfree.chart.plot.PlotOrientation;  
import org.jfree.data.xy.XYSeries;  
import org.jfree.data.xy.XYSeriesCollection;  
import org.jfree.chart.ChartUtilities;
import org.jfree.chart.ChartPanel;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author Ayanami
 */
public class MainApplet extends javax.swing.JApplet {

    /**
     * Initializes the applet MainApplet
     */
    @Override
    public void init() {
        /* Set the Nimbus look and feel */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(MainApplet.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(MainApplet.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(MainApplet.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(MainApplet.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        //</editor-fold>

        /* Create and display the applet */
        try {
            java.awt.EventQueue.invokeAndWait(new Runnable() {
                public void run() {
                    initComponents();
                    resize(1100,600); //Standard Size
                }
            });
        } catch (Exception ex) {
            ex.printStackTrace();
        }
    }

    /**
     * This method is called from within the init() method to initialize the
     * form. WARNING: Do NOT modify this code. The content of this method is
     * always regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        JFrame_Container = new javax.swing.JPanel();
        mainTabbedPanel = new javax.swing.JTabbedPane();
        fileParserJPanel = new javax.swing.JPanel();
        protoDescriptionLabel = new javax.swing.JLabel();
        instructionJLabel = new javax.swing.JLabel();
        methDataInputJPanel = new javax.swing.JPanel();
        fileUpload1JLabel = new javax.swing.JLabel();
        chooseFile1JButton = new javax.swing.JButton();
        fileSelect1JLabel = new javax.swing.JLabel();
        fileSelect1JTextField = new javax.swing.JTextField();
        process1JButton1 = new javax.swing.JButton();
        geneExpDataInputJPanel = new javax.swing.JPanel();
        fileUpload2JLabel1 = new javax.swing.JLabel();
        chooseFile2JButton = new javax.swing.JButton();
        fileSelect2JLabel = new javax.swing.JLabel();
        fileSelect2JTextField = new javax.swing.JTextField();
        process2JButton2 = new javax.swing.JButton();
        visualiserJPanel = new javax.swing.JPanel();
        graphingJPanel = new javax.swing.JPanel();
        dataAnalysisJPanel = new javax.swing.JPanel();
        logJPanel = new javax.swing.JPanel();
        jScrollPane1 = new javax.swing.JScrollPane();
        logJTextArea = new javax.swing.JTextArea();

        setMinimumSize(new java.awt.Dimension(800, 800));

        JFrame_Container.setBorder(javax.swing.BorderFactory.createTitledBorder("EMC2 - Prototype 3"));
        JFrame_Container.setFocusable(false);
        JFrame_Container.setMinimumSize(new java.awt.Dimension(600, 600));

        mainTabbedPanel.setName(""); // NOI18N

        protoDescriptionLabel.setText("EMC2 - Prototype 3");

        instructionJLabel.setText("Instructions: Please blah blah blah");

        methDataInputJPanel.setBorder(javax.swing.BorderFactory.createTitledBorder(javax.swing.BorderFactory.createTitledBorder(""), "Step 1 - Methylation Data Input"));
        methDataInputJPanel.setPreferredSize(new java.awt.Dimension(260, 290));

        fileUpload1JLabel.setText("- File Upload:");

        chooseFile1JButton.setText("Choose File");
        chooseFile1JButton.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                chooseFile1JButtonActionPerformed(evt);
            }
        });

        fileSelect1JLabel.setText("- File Selected:");

        fileSelect1JTextField.setEditable(false);
        fileSelect1JTextField.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                fileSelect1JTextFieldActionPerformed(evt);
            }
        });

        process1JButton1.setText("Process Methylation File");
        process1JButton1.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                process1JButton1ActionPerformed(evt);
            }
        });

        javax.swing.GroupLayout methDataInputJPanelLayout = new javax.swing.GroupLayout(methDataInputJPanel);
        methDataInputJPanel.setLayout(methDataInputJPanelLayout);
        methDataInputJPanelLayout.setHorizontalGroup(
            methDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(methDataInputJPanelLayout.createSequentialGroup()
                .addGroup(methDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addComponent(fileUpload1JLabel)
                    .addComponent(fileSelect1JLabel))
                .addGap(39, 39, 39)
                .addGroup(methDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addComponent(fileSelect1JTextField)
                    .addGroup(methDataInputJPanelLayout.createSequentialGroup()
                        .addGroup(methDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                            .addComponent(process1JButton1)
                            .addComponent(chooseFile1JButton, javax.swing.GroupLayout.PREFERRED_SIZE, 200, javax.swing.GroupLayout.PREFERRED_SIZE))
                        .addGap(0, 122, Short.MAX_VALUE)))
                .addContainerGap())
        );
        methDataInputJPanelLayout.setVerticalGroup(
            methDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(methDataInputJPanelLayout.createSequentialGroup()
                .addGap(16, 16, 16)
                .addGroup(methDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(fileUpload1JLabel)
                    .addComponent(chooseFile1JButton))
                .addGap(27, 27, 27)
                .addGroup(methDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(fileSelect1JLabel, javax.swing.GroupLayout.PREFERRED_SIZE, 22, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(fileSelect1JTextField, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addGap(32, 32, 32)
                .addComponent(process1JButton1)
                .addContainerGap(54, Short.MAX_VALUE))
        );

        geneExpDataInputJPanel.setBorder(javax.swing.BorderFactory.createTitledBorder("Step 2 - Gene Expression Data Input"));
        geneExpDataInputJPanel.setPreferredSize(new java.awt.Dimension(260, 290));
        geneExpDataInputJPanel.setRequestFocusEnabled(false);

        fileUpload2JLabel1.setText("- File Upload:");

        chooseFile2JButton.setText("Choose File");
        chooseFile2JButton.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                chooseFile2JButtonActionPerformed(evt);
            }
        });

        fileSelect2JLabel.setText("- File Selected");

        fileSelect2JTextField.setEditable(false);
        fileSelect2JTextField.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                fileSelect2JTextFieldActionPerformed(evt);
            }
        });

        process2JButton2.setText("Process Gene Expression File");
        process2JButton2.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                process2JButton2ActionPerformed(evt);
            }
        });

        javax.swing.GroupLayout geneExpDataInputJPanelLayout = new javax.swing.GroupLayout(geneExpDataInputJPanel);
        geneExpDataInputJPanel.setLayout(geneExpDataInputJPanelLayout);
        geneExpDataInputJPanelLayout.setHorizontalGroup(
            geneExpDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(geneExpDataInputJPanelLayout.createSequentialGroup()
                .addGroup(geneExpDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addComponent(fileUpload2JLabel1)
                    .addComponent(fileSelect2JLabel))
                .addGap(50, 50, 50)
                .addGroup(geneExpDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addGroup(geneExpDataInputJPanelLayout.createSequentialGroup()
                        .addComponent(fileSelect2JTextField)
                        .addContainerGap())
                    .addGroup(geneExpDataInputJPanelLayout.createSequentialGroup()
                        .addGroup(geneExpDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                            .addComponent(process2JButton2)
                            .addComponent(chooseFile2JButton, javax.swing.GroupLayout.PREFERRED_SIZE, 200, javax.swing.GroupLayout.PREFERRED_SIZE))
                        .addGap(0, 121, Short.MAX_VALUE))))
        );
        geneExpDataInputJPanelLayout.setVerticalGroup(
            geneExpDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(geneExpDataInputJPanelLayout.createSequentialGroup()
                .addGap(16, 16, 16)
                .addGroup(geneExpDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(fileUpload2JLabel1)
                    .addComponent(chooseFile2JButton))
                .addGap(29, 29, 29)
                .addGroup(geneExpDataInputJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(fileSelect2JLabel)
                    .addComponent(fileSelect2JTextField, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addGap(28, 28, 28)
                .addComponent(process2JButton2)
                .addContainerGap(javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
        );

        javax.swing.GroupLayout fileParserJPanelLayout = new javax.swing.GroupLayout(fileParserJPanel);
        fileParserJPanel.setLayout(fileParserJPanelLayout);
        fileParserJPanelLayout.setHorizontalGroup(
            fileParserJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(fileParserJPanelLayout.createSequentialGroup()
                .addGap(29, 29, 29)
                .addGroup(fileParserJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addComponent(instructionJLabel, javax.swing.GroupLayout.PREFERRED_SIZE, 225, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(protoDescriptionLabel, javax.swing.GroupLayout.PREFERRED_SIZE, 498, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addGroup(fileParserJPanelLayout.createSequentialGroup()
                        .addComponent(methDataInputJPanel, javax.swing.GroupLayout.PREFERRED_SIZE, 450, javax.swing.GroupLayout.PREFERRED_SIZE)
                        .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                        .addComponent(geneExpDataInputJPanel, javax.swing.GroupLayout.PREFERRED_SIZE, 450, javax.swing.GroupLayout.PREFERRED_SIZE)))
                .addContainerGap(23, Short.MAX_VALUE))
        );
        fileParserJPanelLayout.setVerticalGroup(
            fileParserJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(fileParserJPanelLayout.createSequentialGroup()
                .addContainerGap()
                .addComponent(protoDescriptionLabel, javax.swing.GroupLayout.PREFERRED_SIZE, 26, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                .addComponent(instructionJLabel, javax.swing.GroupLayout.PREFERRED_SIZE, 22, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                .addGroup(fileParserJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING, false)
                    .addComponent(geneExpDataInputJPanel, javax.swing.GroupLayout.DEFAULT_SIZE, 218, Short.MAX_VALUE)
                    .addComponent(methDataInputJPanel, javax.swing.GroupLayout.DEFAULT_SIZE, 218, Short.MAX_VALUE))
                .addContainerGap(228, Short.MAX_VALUE))
        );

        mainTabbedPanel.addTab("File Parser & Settings", fileParserJPanel);

        graphingJPanel.setBorder(javax.swing.BorderFactory.createTitledBorder(javax.swing.BorderFactory.createEtchedBorder(), "Graph Plotter"));
        graphingJPanel.setCursor(new java.awt.Cursor(java.awt.Cursor.DEFAULT_CURSOR));
        graphingJPanel.setMinimumSize(new java.awt.Dimension(650, 340));

        javax.swing.GroupLayout graphingJPanelLayout = new javax.swing.GroupLayout(graphingJPanel);
        graphingJPanel.setLayout(graphingJPanelLayout);
        graphingJPanelLayout.setHorizontalGroup(
            graphingJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGap(0, 864, Short.MAX_VALUE)
        );
        graphingJPanelLayout.setVerticalGroup(
            graphingJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGap(0, 453, Short.MAX_VALUE)
        );

        javax.swing.GroupLayout visualiserJPanelLayout = new javax.swing.GroupLayout(visualiserJPanel);
        visualiserJPanel.setLayout(visualiserJPanelLayout);
        visualiserJPanelLayout.setHorizontalGroup(
            visualiserJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(visualiserJPanelLayout.createSequentialGroup()
                .addGap(44, 44, 44)
                .addComponent(graphingJPanel, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addContainerGap(42, Short.MAX_VALUE))
        );
        visualiserJPanelLayout.setVerticalGroup(
            visualiserJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(visualiserJPanelLayout.createSequentialGroup()
                .addGap(35, 35, 35)
                .addComponent(graphingJPanel, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addContainerGap(12, Short.MAX_VALUE))
        );

        mainTabbedPanel.addTab("Data Visualiser", visualiserJPanel);

        javax.swing.GroupLayout dataAnalysisJPanelLayout = new javax.swing.GroupLayout(dataAnalysisJPanel);
        dataAnalysisJPanel.setLayout(dataAnalysisJPanelLayout);
        dataAnalysisJPanelLayout.setHorizontalGroup(
            dataAnalysisJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGap(0, 962, Short.MAX_VALUE)
        );
        dataAnalysisJPanelLayout.setVerticalGroup(
            dataAnalysisJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGap(0, 522, Short.MAX_VALUE)
        );

        mainTabbedPanel.addTab("Data Analysis", dataAnalysisJPanel);

        logJTextArea.setEditable(false);
        logJTextArea.setColumns(20);
        logJTextArea.setRows(5);
        jScrollPane1.setViewportView(logJTextArea);

        javax.swing.GroupLayout logJPanelLayout = new javax.swing.GroupLayout(logJPanel);
        logJPanel.setLayout(logJPanelLayout);
        logJPanelLayout.setHorizontalGroup(
            logJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(logJPanelLayout.createSequentialGroup()
                .addGap(45, 45, 45)
                .addComponent(jScrollPane1, javax.swing.GroupLayout.PREFERRED_SIZE, 760, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addContainerGap(157, Short.MAX_VALUE))
        );
        logJPanelLayout.setVerticalGroup(
            logJPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(logJPanelLayout.createSequentialGroup()
                .addGap(32, 32, 32)
                .addComponent(jScrollPane1, javax.swing.GroupLayout.PREFERRED_SIZE, 380, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addContainerGap(110, Short.MAX_VALUE))
        );

        mainTabbedPanel.addTab("Log (Only in prototype mode)", logJPanel);

        javax.swing.GroupLayout JFrame_ContainerLayout = new javax.swing.GroupLayout(JFrame_Container);
        JFrame_Container.setLayout(JFrame_ContainerLayout);
        JFrame_ContainerLayout.setHorizontalGroup(
            JFrame_ContainerLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(JFrame_ContainerLayout.createSequentialGroup()
                .addGap(42, 42, 42)
                .addComponent(mainTabbedPanel, javax.swing.GroupLayout.PREFERRED_SIZE, 967, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addContainerGap(javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
        );
        JFrame_ContainerLayout.setVerticalGroup(
            JFrame_ContainerLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(JFrame_ContainerLayout.createSequentialGroup()
                .addContainerGap()
                .addComponent(mainTabbedPanel, javax.swing.GroupLayout.PREFERRED_SIZE, 550, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addContainerGap(205, Short.MAX_VALUE))
        );

        javax.swing.GroupLayout layout = new javax.swing.GroupLayout(getContentPane());
        getContentPane().setLayout(layout);
        layout.setHorizontalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup()
                .addComponent(JFrame_Container, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE)
                .addContainerGap())
        );
        layout.setVerticalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(javax.swing.GroupLayout.Alignment.TRAILING, layout.createSequentialGroup()
                .addComponent(JFrame_Container, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE)
                .addContainerGap())
        );
    }// </editor-fold>//GEN-END:initComponents

    private void chooseFile1JButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_chooseFile1JButtonActionPerformed
        //int returnVal = fc.showOpenDialog(MainApplet.this);
        //if(returnVal == JFileChooser.APPROVE_OPTION) { //If chosen
         //   File methDataFile = fc.getSelectedFile();
         //   fileSelect1JTextField.setText(methDataFile.getName());
         //   logJTextArea.append("Selected: " + methDataFile.getName() + ".\n" );
       // } else {
            logJTextArea.append("Action cancelled by user.\n");
        //}
    }//GEN-LAST:event_chooseFile1JButtonActionPerformed

    private void fileSelect1JTextFieldActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_fileSelect1JTextFieldActionPerformed
        // TODO add your handling code here:
    }//GEN-LAST:event_fileSelect1JTextFieldActionPerformed

    private void process1JButton1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_process1JButton1ActionPerformed
        if(fileSelect1JTextField.getText().isEmpty() == false) {
            logJTextArea.append("Processing Meth Data file.\n");
        }

        /**
        * ISOLATED COMMAND TO LAUNCH CHART!
        * IMPORTANT - LAUNCH ON CLICK
        */
        
        XYSeriesCollection dataset = new XYSeriesCollection();
        XYSeries xySeries1 = new XYSeries( "Methylation" );
        xySeries1.add( 1, 10 );
        xySeries1.add( 2, 4 );
        xySeries1.add( 3, 6 );
        xySeries1.add( 4, 12 );
        xySeries1.add( 5, 11 );
        xySeries1.add( 6, 29 );
        xySeries1.add( 7, 33 );

        XYSeries xySeries2 = new XYSeries( "SNP" );
        xySeries2.add( 1, 1 );
        xySeries2.add( 2, 2 );
        xySeries2.add( 3, 18 );
        xySeries2.add( 4, 5 );
        xySeries2.add( 5, 52 );
        xySeries2.add( 6, 66 );
        xySeries2.add( 7, 83 );

        dataset.addSeries( xySeries1 );
        dataset.addSeries( xySeries2 );

        JFreeChart tempChart = ChartFactory.createXYLineChart( "Test Gene/Meth Graph",
            "Meth",  //X Axis
            "SNP", //Y Axis
            dataset,
            PlotOrientation.VERTICAL,
            true,
            false,
            false );

        ChartPanel finalChart = new ChartPanel(tempChart,true); //Required to bind it to a JFrame/JPanel

        /**
        * This section is used for the implementation of buttons to assist with
        * the navigation of the chart.
        */
        JButton zoomOutButton = new JButton("Zoom Out All");
        JButton scrollRightButton = new JButton("Scroll Right");
        JButton scrollLeftButton = new JButton("Scroll Left");

        /**
        * This section is to insert the graph into the Data Visualiser tab
        * AddComponent = Adding graph into the page, param indicate size of the chart
        * The buttons are all drawn dependent on the data provided
        * PLEASE DO NOT EDIT UNLESS YOU KNOW WHAT YOU ARE DOING
        */

        javax.swing.GroupLayout graphingJPanelLayout = new javax.swing.GroupLayout(graphingJPanel);
        graphingJPanel.setLayout(graphingJPanelLayout);
        graphingJPanelLayout.setHorizontalGroup(
            graphingJPanelLayout.createParallelGroup(javax.swing.
                GroupLayout.Alignment.LEADING)
            .addGroup(graphingJPanelLayout.createSequentialGroup()
                //.addGap(50,50,50)
                //.addComponent(finalChart, javax.swing.GroupLayout.PREFERRED_SIZE, 650, javax.swing.GroupLayout.PREFERRED_SIZE))
            .addComponent(finalChart, 850, 850, 850))
        .addGroup(graphingJPanelLayout.createSequentialGroup()
            .addGap(350,350,350) //Gap addition(min, preffered, Max)
            .addComponent(zoomOutButton))
        .addGroup(graphingJPanelLayout.createSequentialGroup() //Scroll RIGHT
            .addGap(455,455,455)
            .addComponent(scrollRightButton))
        .addGroup(graphingJPanelLayout.createSequentialGroup() //Scroll LEFT
            .addGap(255,255,255)
            .addComponent(scrollLeftButton))
        //.addContainerGap(50, Short.MAX_VALUE))
        );
        graphingJPanelLayout.setVerticalGroup(
            graphingJPanelLayout.createParallelGroup(javax.swing.
                GroupLayout.Alignment.BASELINE)
            .addGroup(graphingJPanelLayout.createSequentialGroup()
                //.addGap(-30,-30,-30)
                .addComponent(finalChart, 340, 340, 340))
            .addGroup(graphingJPanelLayout.createSequentialGroup()
                .addGap(360,360,360)
                .addComponent(zoomOutButton))
            .addGroup(graphingJPanelLayout.createSequentialGroup() //Scroll RIGHT
                .addGap(360,360,360)
                .addComponent(scrollRightButton))
            .addGroup(graphingJPanelLayout.createSequentialGroup() //Scroll LEFT
                .addGap(360,360,360)
                .addComponent(scrollLeftButton))
            //.addContainerGap(50, Short.MAX_VALUE))
        );

        /**************************************************************************/
    }//GEN-LAST:event_process1JButton1ActionPerformed

    private void chooseFile2JButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_chooseFile2JButtonActionPerformed
        //int returnVal = fc.showOpenDialog(MainApplet.this);
        //if(returnVal == JFileChooser.APPROVE_OPTION) { //If chosen
         //   File methDataFile = fc.getSelectedFile();
          //  fileSelect2JTextField.setText(methDataFile.getName());
           // logJTextArea.append("Selected: " + methDataFile.getName() + ".\n" );
        //} else {
         //   logJTextArea.append("Action cancelled by user.\n");
        //}
    }//GEN-LAST:event_chooseFile2JButtonActionPerformed

    private void fileSelect2JTextFieldActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_fileSelect2JTextFieldActionPerformed
        // TODO add your handling code here:
    }//GEN-LAST:event_fileSelect2JTextFieldActionPerformed

    private void process2JButton2ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_process2JButton2ActionPerformed
        if(fileSelect2JTextField.getText().isEmpty() == false) {
            logJTextArea.append("Processing Gene Data file.\n");
        }
    }//GEN-LAST:event_process2JButton2ActionPerformed

    //Create File Chooser
    //JFileChooser fc = new JFileChooser();
    
    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JPanel JFrame_Container;
    private javax.swing.JButton chooseFile1JButton;
    private javax.swing.JButton chooseFile2JButton;
    private javax.swing.JPanel dataAnalysisJPanel;
    private javax.swing.JPanel fileParserJPanel;
    private javax.swing.JLabel fileSelect1JLabel;
    private javax.swing.JTextField fileSelect1JTextField;
    private javax.swing.JLabel fileSelect2JLabel;
    private javax.swing.JTextField fileSelect2JTextField;
    private javax.swing.JLabel fileUpload1JLabel;
    private javax.swing.JLabel fileUpload2JLabel1;
    private javax.swing.JPanel geneExpDataInputJPanel;
    private javax.swing.JPanel graphingJPanel;
    private javax.swing.JLabel instructionJLabel;
    private javax.swing.JScrollPane jScrollPane1;
    private javax.swing.JPanel logJPanel;
    private javax.swing.JTextArea logJTextArea;
    private javax.swing.JTabbedPane mainTabbedPanel;
    private javax.swing.JPanel methDataInputJPanel;
    private javax.swing.JButton process1JButton1;
    private javax.swing.JButton process2JButton2;
    private javax.swing.JLabel protoDescriptionLabel;
    private javax.swing.JPanel visualiserJPanel;
    // End of variables declaration//GEN-END:variables
}
